llvm/lib/IR/ConstantRangeList.cpp

```diff
@@ -81,6 +81,97 @@ void ConstantRangeList::insert(const ConstantRange &NewRange) {
   }
 }
 
+ConstantRangeList
+ConstantRangeList::unionWith(const ConstantRangeList &CRL) const {
+  assert(getBitWidth() == CRL.getBitWidth() &&
+         "ConstantRangeList bitwidths don't agree!");
+  // Handle common cases.
+  if (empty())
+    return CRL;
+  if (CRL.empty())
+    return *this;
+
+  ConstantRangeList Result;
+  size_t i = 0, j = 0;
+  // "PreviousRange" tracks the unioned range (its lower is fixed
+  // and the upper may be updated over iterations).
+  // If "PreviousRange" cannot contain a new unioned range, push it
```

Comment:
nit: might not need to mention the "If ... cannot contain a new ...", since it's elaborated below where it happens.

For above, could emphasize that ""PreviousRange" tracks the *lowest* unioned range *that is being processed*..." ?

Reply:
Done.